go-controller/pkg/cni/helper_linux_test.go

```diff
@@ -1651,6 +1652,21 @@ func TestConfigureOVS_getPfEncapIpWithError(t *testing.T) {
 	}
 }
 
+func TestSetupIngressFilter(t *testing.T) {
```

Comment:
thanks for the test!

Reply:
I think is better to have an e2e, I don't see this adding much value.

Reply:
I don't see them as mutually exclusive tbh. Why don't you add and e2e test and keep this one around ?

Reply:
I see the unit test quite intrussive since we need to pass the the testing function the table and the lifetime field matcher, at the end to just check the nftables rule is as expected, I know we can make it better by having two functions, on is the one we are going to unit test and the other is the production code one that do not need to pass the table and the matcher.

Reply:
Knowing you're generating the nftable rule you expect is valuable enough IMHO. We can discuss the function's signature though. 

I won't pretend I understood what you meant with "we can make it better by having two functions".

Reply:
Ok, I will put the unit test back with some refactoring so production code do not have to pass the table and the matcher.

Reply:
Thank you, it does make sense to me.